### PR TITLE
[SPARK-38332][SQL] Add the `DATEADD()` and `DATE_ADD()` aliases for `TIMESTAMPADD()`

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to add two aliases for the `TIMESTAMPADD()` function introduced by https://github.com/apache/spark/pull/35502:
- `DATEADD()`
- `DATE_ADD()`

### Why are the changes needed?
1. To make the migration process from other systems to Spark SQL easier.
2. To achieve feature parity with other DBMSs.

### Does this PR introduce _any_ user-facing change?
No. The new aliases just extend Spark SQL API.

### How was this patch tested?
1. By running the existing test suites:
```
$ build/sbt "test:testOnly *SQLKeywordSuite"
```
3. and new checks:
```
$ build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite -- -z date.sql"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@srielau @entong @superdupershant Any feedback is welcome.

**Review by HyukjinKwon — APPROVED** [review]
I'm fine w/ this. Look fine.

File: sql/core/src/test/resources/sql-tests/results/ansi/date.sql.out [hunk 1]
```diff
@@ -660,3 +660,83 @@ struct<>
...
+-- !query
+select dateadd(HOUR, -1, timestamp'2022-02-25 01:02:03')
+-- !query schema
+struct<timestampadd(HOUR, -1, TIMESTAMP '2022-02-25 01:02:03'):timestamp>
+-- !query output
+2022-02-25 00:02:03
+
+
+-- !query
+select date_add(DAY, 367, date'2022-02-25')
```
Comment (superdupershant) [inline sql/core/src/test/resources/sql-tests/results/ansi/date.sql.out:706]: I think the convention is if the third argument is of type DATE the expression's return type is:

DATE if the unit DAY or larger
TIMESTAMP if the unit is smaller than DAY i.e. (HOUR, MINUTE, SECOND, etc...)

**Review by superdupershant — CHANGES_REQUESTED** [review]
Looks good except the return type shouldn't always be TIMESTAMP if the input expression and operation fit within a DATE type.
Sorry for missing this in the original TIMESTAMPADD change as well.

File: sql/core/src/test/resources/sql-tests/results/ansi/date.sql.out (hunk 1 above)
Comment (MaxGekk) [reply sql/core/src/test/resources/sql-tests/results/ansi/date.sql.out:706]: I don't think we should do such kind of changes in the PR which just introduces aliases. Let's do that separately w/ tests if the feature is supported by other DBMSs as well.

Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/results/ansi/date.sql.out:706]: Do you mean timestampadd can take date inputs but dateadd can't take timestamp inputs? Then it's not a simple alias any more.

Comment (superdupershant) [reply sql/core/src/test/resources/sql-tests/results/ansi/date.sql.out:706]: @cloud-fan no both expressions can take date and timestamp inputs it is just a simple alias.

@MaxGekk yes I agree fixing it in a separate PR makes sense.

**Review by superdupershant — APPROVED** [review]
LGTM let's just fix the return types in a separate PR.

**Review by cloud-fan — APPROVED** [review]

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @HyukjinKwon @superdupershant @cloud-fan @gengliangwang for review.

**Pull request closed without merging**